app/controllers/api/maintenance_stats_controller.rb

```diff
@@ -19,6 +19,30 @@ def begin_watching_bulk
     head :accepted
   end
 
+  def begin_watching_repositories
+    lookup_names = params.require(:repositories).map { |p| p.permit(%i[host_type full_name]).to_h.symbolize_keys }
+
+    supported_host_types = ["github"]
+
+    # we only support GitHub so grab a working token now so we
+    # don't run into issues trying to find one in the loop for
+    # each repository name
+    auth_token = AuthToken.find_token(:v3)
+
+    repos = lookup_names.map do |repo_host_and_name|
+      return render json: { error: "#{repo_host_and_name[:host_type]} is not a supported host" }, status: :bad_request unless supported_host_types.include?(repo_host_and_name[:host_type].downcase)
+
+      Repository.create_from_host(repo_host_and_name[:host_type], repo_host_and_name[:full_name], auth_token.token)
+    end
+
+    # compact repos array to remove any invalid/not found repositories
+    repos.compact.each do |repo|
```

Comment:
This decision impacts what happens when we cannot find a repository by name. The code now basically just ignores them and queues up stats for the repositories we did find. This means if we have invalid repository names we won't get an error back from this API nor will it start retrieving stats for them.